docs/Advanced-Usage/Heterogeneous-SoCs.rst

```diff
@@ -0,0 +1,127 @@
+Heterogeneous SoCs
+===============================
+
+The Chipyard framework involves multiple cores and accelerators that can be composed in arbitrary ways.
+This discussion will focus on how you combine Rocket, BOOM and Hwacha in particular ways to create a unique SoC.
+
+Creating a Rocket and BOOM System
+-------------------------------------------
+
+Instantiating an SoC with Rocket and BOOM cores is all done with the configuration system and two specific mixins.
+Both BOOM and Rocket have mixins labelled ``WithNBoomCores(X)`` and ``WithNBigCores(X)`` that automatically create ``X`` copies of the core.
+When used together you can create a heterogeneous system.
+The following example shows a dual core BOOM with a single core Rocket.
+
+.. code-block:: scala
+
+    class DualBoomAndOneRocketConfig extends Config(
+      new WithNormalBoomRocketTop ++
+      new WithBootROM ++
+      new boom.system.WithRenumberHarts ++
+      new boom.common.WithRVC ++
+      new boom.common.DefaultBoomConfig ++
+      new boom.system.WithNBoomCores(2) ++
+      new freechips.rocketchip.subsystem.WithoutTLMonitors ++
+      new freechips.rocketchip.subsystem.WithNBigCores(1) ++
+      new freechips.rocketchip.system.BaseConfig)
+
+In this example, the ``WithN...Cores(X)`` mixins set up particular parameters for the duplicated core.
```

Comment:
This sentence might be a bit confusing. Does this imply default parameters? If so, why do you need the additional (DefaultBOOMConfig mixin)?

Reply:
Hmm. So this is just changed in BOOM land. Now, you have to have both mixins (aka. the With... and the XTypeConfig). I can reword this to something like: "In this example, the with "WithN...Cores(Z)" sets up the default parameters needed to have Z amount of cores. For something like BOOM, we add an extra mixin called "XYZ" that is used to change the parameters to a different size core."